storage/rocksdb/ha_rocksdb.cc

```diff
@@ -4238,6 +4276,68 @@ bool ha_rocksdb::init_with_fields() {
   DBUG_RETURN(false); /* Ok */
 }
 
+
+/*
+  If the key is a TTL key, we may need to filter it out.
+
+  The purpose of read filtering for tables with TTL is to ensure that
+  during a transaction a key which has expired already but not removed by
+  compaction yet is not returned to the user.
+
+  Without this the user might be hit with problems such as disappearing
+  rows within a transaction, etc, because the compaction filter ignores
+  snapshots when filtering keys.
+*/
+int ha_rocksdb::should_hide_ttl_rec(const rocksdb::Slice &ttl_rec_val,
+                                    const int64_t curr_ts) const {
+  DBUG_ASSERT(m_pk_descr != nullptr);
+  DBUG_ASSERT(m_pk_descr->has_ttl());
+
+  /*
+    Curr_ts can only be 0 if there are no snapshots open.
+    should_hide_ttl_rec can only be called when there is >=1 snapshots, unless
+    we are filtering on the write path (single INSERT/UPDATE) in which case
+    we are passed in the current time as tx->m_ttl_filter_timestamp.
+  */
+  if (curr_ts == 0) {
+    sql_print_error("MyRocks: No snapshot open or timestamp passed in call to "
+                    "should_hide_ttl_rec, Aborting.");
+    abort_with_stack_traces();
```

Comment:
I think you are correct and we can make do without `m_ttl_filter_timestamp`.

in `check_uniqueness_and_lock()`
   PK found, and it's not expired => return HA_ERR_DUP_KEY
   PK found, and it IS expired => pretend not found, overwrite it
   SK found, then PK must also be expired because it was checked first => pretend not found, overwrite

So I don't think there will be any problem with using `std::time()`.  I'd still like to keep the `curr_ts == 0` check though, to make sure there aren't any unexpected cases where there aren't any snapshots open, think that's ok?